packages/lodestar/src/chain/genesis/genesis.ts

```diff
@@ -49,14 +138,66 @@ export function initializeBeaconStateFromEth1(
     )
   );
 
+  applyTimestamp(config, state, eth1Timestamp);
+
   // Process deposits
+  applyDeposits(config, state, deposits);
+
+  return state as TreeBacked<BeaconState>;
+}
+
+/**
+ * Apply eth1 block hash to state.
+ * @param config IBeaconConfig
+ * @param state BeaconState
+ * @param eth1BlockHash eth1 block hash
+ */
+function applyEth1BlockHash(config: IBeaconConfig, state: BeaconState, eth1BlockHash: Bytes32): void {
+  state.eth1Data.blockHash = eth1BlockHash;
+  state.randaoMixes = Array<Bytes32>(config.params.EPOCHS_PER_HISTORICAL_VECTOR).fill(eth1BlockHash);
+}
+
+/**
+ * Apply eth1 block timestamp to state.
+ * @param config IBeaconState
+ * @param state BeaconState
+ * @param eth1Timestamp eth1 block timestamp
+ */
+function applyTimestamp(config: IBeaconConfig, state: BeaconState, eth1Timestamp: number): void {
+  state.genesisTime =
+    eth1Timestamp - eth1Timestamp % config.params.MIN_GENESIS_DELAY + 2 * config.params.MIN_GENESIS_DELAY;
+}
+
+/**
+ * Apply deposits to state.
+ * @param config IBeaconConfig
+ * @param state BeaconState
+ * @param deposits full list of deposits
+ */
+function applyDeposits(
```

Comment:
could you move those util functions to separate file so we keep codeclimate happy